### DISCUSSION TITLE
v5.7.0 User Preview

### DISCUSSION THREAD

**Discussion by xiaokangwang** [v2fly/v2ray-core#2549] [General]
## New Features
* **meek**: a new transport protocol meek is added.
  * [CN](https://www.v2fly.org/v5/config/stream/meek.html) [EN](https://www.v2fly.org/en_US/v5/config/stream/meek.html) Documents have been updated.
  * Meek is a censorship resistant protocol that encode traffic into plain HTTP request/responses. This would allow the traffic to be reflected over significant amount of platforms to migrate the effect of IP blocking.
  * The transfer performance for meek is low, and intended for situation where other methods are not realible enough.
  * Thanks [David Fifield](https://www.bamsoftware.com/software/index.html), the developer of meek's original implementation.

<hr /><em>This discussion was created from the release <a href='https://github.com/v2fly/v2ray-core/releases/tag/v5.7.0'>v5.7.0 User Preview</a>.</em>

**Comment by computerscot** [comment]
What is the correct format for the configuration file for meek? For example:

```
            "streamSettings": {
                "network": "meek",
                "url": "www.cscot.buzz"
            }
```

Produces error message:

```
Failed to start: main/commands: failed to load config: [/usr/local/etc/v2ray/config.json] > infra/conf/v4: Config: unknown transport protocol: meek
```

**Reply by xiaokangwang** [reply]
Your configuration is incorrect:
1. meek is only supported in jsonv5 configuration format you are using v4
2. here an example: https://gist.github.com/xiaokangwang/b8f3a77780cae99eba68e7b127522883

**Comment by computerscot** [comment]
That works, thanks. Completed example is at https://cscot.pages.dev/2023/05/31/v2ray-meek/

**Comment by epsilonzeio** [comment]
Good job! Thanks
Just a question: Does meek protocol fix V2ray crashing in Iran due to the last DPI update's (every week) both from MCI and MTN operators?
